### DISCUSSION TITLE
Solicitation Request json.decoder.JSONDecodeError

### DISCUSSION THREAD

**Discussion by marek715** [saleweaver/python-amazon-sp-api#243] [Q&A] [answered]
Hello!

Can someone help show me how I can setup a set_trace to capture more information on the following json.decoder.JSONDecodeError?  (full details described in this issue - link below)
[/issues/238]

I would be happy to help debug/fix, but not certain how I can setup a set_trace to capture the information Amazon is sending that frequently causes decode errors here:
`File "C:\Users\user\anaconda3\envs\py36\lib\json\decoder.py", line 355, in raw_decode raise JSONDecodeError("Expecting value", s, err.value) from None json.decoder.JSONDecodeError: Expecting value: line 1 column 1`

It happens once every 20-50 orders, so it's fairly reproducible, and it doesn't happen on every solicitation request since most go through without an issue.  Currently, I am restarting the program and keeping a log of where it left off.  A re-send on the same order ID always works.    

Thanks in advance,
Marek

Initial Edits: fix bug report hyperlink

**Comment by saleweaver** [comment, marked as answer]
You can log the response as text before this line: 

https://github.com/saleweaver/python-amazon-sp-api/blob/d89d68a3d0662599b651be8a6a1dda7f0d2c8b7a/sp_api/base/client.py#L109

Then you should be able to see if something actually is wrong with the response once one fails.
